src/Router.ts

```diff
@@ -439,8 +439,36 @@ export class Router<C extends Context> extends Evented implements RouterInterfac
 		return this._outletContextMap.has(outletId);
 	}
 
-	getOutlet(outletId: string): OutletContext | undefined {
-		return this._outletContextMap.get(outletId);
+	getOutlet(outletIds: string | string[]): OutletContext | undefined {
+		if (Array.isArray(outletIds)) {
+			const combinedParams = outletIds.reduce((params, cur, index) => {
```

Comment:
This section feels quite complicated now, do you think it could be simplified slightly? We could maybe use the path to determine if a matching outlet is deeper than an existing match and not need to loop through the context map.

Rough example:

```ts
let matchingOutlet: OutletContext | undefined = undefined;
let matchingParams: Parameters = {};
let matchingLocation = '';
for (let i = 0; i < outletIds.length; i++) {
	const outletContext = this._outletContextMap.get(outletIds[i]);
	if (outletContext) {
		const { params, location } = outletContext;
		matchingParams = { ...matchingParams, ...params };
		if (!matchingOutlet || matchingLocation.indexOf(location) === -1) {
			matchingLocation = location;
			matchingOutlet = {
				...outletContext,
				params: matchingParams
			};
		}
	}
}
return matchingOutlet;
```

Reply:
Thank you! I've just added a `const outletIds = Array.isArray(outletId) ? outletId : [outletId];` to make sure we're dealing with an array, but apart from that, I'm using your example 👍